# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/script/ScriptLoadRequest.h
# Commit: a71a2a98b6bc
# Full Hash: a71a2a98b6bc017a7bb60d4b7c41f543f156eede
# Author: Yulia Startsev <ystartsev@mozilla.com>
# Date: 2022-02-21 21:39:46
# Regressor Bug: 1742437
# File Overlap Count: 1
# Description:
#   Bug 1742437 - move mElement from ScriptFetchOptions to DOMScriptLoadContext; r=jonco,smaug
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D136742
# ==============================================================================

diff -r d52858ae58d2 -r a71a2a98b6bc dom/script/ScriptLoadRequest.h
--- a/dom/script/ScriptLoadRequest.h	Mon Feb 21 15:39:04 2022 +0000
+++ b/dom/script/ScriptLoadRequest.h	Mon Feb 21 15:39:05 2022 +0000
@@ -59,7 +59,7 @@
   NS_DECL_CYCLE_COLLECTION_NATIVE_CLASS(ScriptFetchOptions)
 
   ScriptFetchOptions(mozilla::CORSMode aCORSMode,
-                     enum ReferrerPolicy aReferrerPolicy, Element* aElement,
+                     enum ReferrerPolicy aReferrerPolicy,
                      nsIPrincipal* aTriggeringPrincipal,
                      nsIGlobalObject* aWebExtGlobal);
 
@@ -341,7 +341,7 @@
   virtual ~DOMScriptLoadContext();
 
  public:
-  explicit DOMScriptLoadContext(ScriptLoadRequest* aRequest);
+  explicit DOMScriptLoadContext(Element* aElement, ScriptLoadRequest* aRequest);
 
   NS_DECL_CYCLE_COLLECTING_ISUPPORTS
   NS_DECL_CYCLE_COLLECTION_SCRIPT_HOLDER_CLASS(DOMScriptLoadContext)
@@ -456,6 +456,9 @@
   Atomic<Runnable*> mRunnable;  // Runnable created when dispatching off thread
                                 // compile. Tracked here so that it can be
                                 // properly released during cancellation.
+
+  nsCOMPtr<Element> mElement;
+
   RefPtr<ScriptLoadRequest> mRequest;
 
   // Non-null if there is a document that this request is blocking from loading.